src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.SemanticChangedEventSource.cs

```diff
@@ -61,15 +61,16 @@ private void OnSubjectBufferChanged(object sender, TextContentChangedEventArgs e
             private void OnOpenedDocumentSemanticChanged(object sender, Document document)
             {
                 var workspace = this.CurrentWorkspace;
+                var solution = document.Project.Solution;
 
-                if (document.Project.Solution.Workspace != workspace)
+                if (solution.Workspace != workspace)
                 {
                     return;
                 }
 
-                var documentIds = workspace.GetRelatedDocumentIds(SubjectBuffer.AsTextContainer());
-
-                if (!documentIds.Contains(document.Id))
+                var documentId = workspace.GetDocumentIdInCurrentContext(SubjectBuffer.AsTextContainer());
+                var relatedDocumentIds = solution.GetRelatedDocumentIds(documentId);
+                if (!relatedDocumentIds.Contains(document.Id))
```

Comment:
Ditto as the other tagger provider: this is changing behavior in a subtle way which I don't _think_ matters but could somehow.